src/Dev/SapphireTest.php

```diff
@@ -546,6 +549,8 @@ protected function tearDown()
 
         // Call state helpers
         static::$state->tearDown($this);
+
+        static::$tempDB->rollbackTransaction();
```

Comment:
The rollbackTransaction() needs to be in an if() statement that matches the if() on setUp().

`if ($this->shouldSetupDatabaseForCurrentTest($fixtureFiles)) {`

Otherwise you're rolling back more than you're starting transaction.